nn/layer.py

```diff
@@ -32,6 +44,19 @@ def __init__(self, units: int):
         self._dw = None
         self._db = None
 
+    # Add getter and setter methods for weights and bias
```

Comment:
Thanks for the great additions. I'm wondering why are you not using property decorator for the getter and setters?

e.g.

```python

@property
def weights(self):
  return self._weights
  
@weights.setter
def weights(self, value):
  self._weights = value
```

Reply:
Yes you was right, I have made the necessary changes as it was the good practice. Also I have explicitly defined datatypes of parameters for the setters of weights and bias. Did you get a change to run the code?

Reply:
Re: running the code, I've added another comment for you to look at - you haven't updated the optimisers after adding property based getter/ setters.